### ISSUE TITLE
Profiling Rake Tasks launched with Cron

### ISSUE THREAD

**Issue by ishields** [getsentry/sentry#63559]
### Environment

SaaS (https://sentry.io/)

### What are you trying to accomplish?

I am trying to profile Ruby/Rails rake tasks that run with Cron so that I can have insight into their performance (slow queries, n+1 issues, etc) like I do with normal rails transactions. I use a cron file run rake tasks. Ex:
```
0 0,2,4,6,8,10,12,14,16,18,20,22 * * * /bin/bash -l -c 'cd /home/deploy/projectName/releases/20240... && RAILS_ENV=production RAILS_ENV=production bundle exec rake environment myraketasks:hello_workd --silent >> log/cron.log 2>> log/cron-error.log'
```

It seems that even if I put profile sampling to 100%, these rake tasks are never picked up and I have no visibility into how they're performing.  

### How are you getting stuck?

I'm not sure how to enable profiling for these rake tasks so that I can monitor their performance.

### Where in the product are you?

Performance

**Comment by k-fish** [comment]
Hey there @ishields, it sounds like you might not have integrations setup correctly if you are seeing transactions but not database queries etc. which should be working if you are making database queries through `ActiveRecord`. 

Double check that you have the correct gems installed (see [automatic instrumentation](https://docs.sentry.io/platforms/ruby/performance/instrumentation/automatic-instrumentation/) in the docs)

If you're still having trouble you can post your config / init setup, or otherwise try asking for more ruby specific help in our https://github.com/getsentry/sentry-ruby sdk repo.

**Comment by ishields** [comment]
Thanks for your response. I think I have the right integrations setup because in the main rails app and in sidekiq everything behaves correctly. It's really just the rake tasks (launched via cron) where this is an issue.

Here is my configuration:
```
Sentry.init do |config|
    puts 'Initializing Sentry'

    config.dsn = '.....'
    config.breadcrumbs_logger = %i[active_support_logger http_logger]
    config.enable_tracing = true

    # Set traces_sample_rate to 1.0 to capture 100%
    # of transactions for performance monitoring.
    # We recommend adjusting this value in production.
    config.traces_sample_rate = 1
    config.profiles_sample_rate = 1
    config.sample_rate = 1

    config.enabled_environments = %w[development production]
ScheduleJobsWorker].map(&:downcase)
    config.traces_sampler =
      lambda do |context|
        LogUtil.info("Sentry Trace #{context}")
        return true
      end
  end
```

```
# sentry
gem 'stackprof' #use for sentry
gem "sentry-ruby"
gem "sentry-rails"
gem "sentry-sidekiq"
```

I put some print statements in my config below - when running the rake tasks it prints 'Initializing Sentry' but doesn't print "Sentry Trace #{context}".  The only way I managed to get it to print Sentry Trace is by manually creating a transaction as shown in my original example.

**Comment by k-fish** [comment]
> I think I have the right integrations setup because in the main rails app and in sidekiq everything behaves correctly

Just to clarify, you are seeing database spans etc. in regular rails usage, just not rake?

It sounds like it might be an sdk specific issue, I'm going to tag the sdk team to help you look further.

**Comment by ishields** [comment]
Yep, that's correct I see database spans etc in regular rails usage (the webserver and the sidekiq tasks). This issue is only happening for me with  rake tasks. Thanks!
